src/persist-client/README.md

```diff
@@ -0,0 +1,87 @@
+# Persist
+
+Persist is an implementation detail of STORAGE. Its "public" API is used only
+by STORAGE and code outside of STORAGE should be talking to STORAGE, not
+persist. However, having this strong API boundary between the two allows for the
+two teams to execute independently.
+
+Persist's primary abstraction is a "shard", which is a durable and definite
+[Time-Varying Collection (TVC)]. Persist requires that the collection's "data"
+is key-value structured (a `()` unit value is fine) but otherwise allows the
+key, value, time, and diff to be abstract in terms of the [Codec] and [Codec64]
+encode/decode traits. As a result, persist is independent of Materialize's
+internal data formats: `Row` etc.
+
+[Time-Varying Collection (TVC)]: https://github.com/MaterializeInc/materialize/blob/13603a6ee1c13eaa243948fc6a99e8fe2a9d0cdd/doc/developer/platform/formalism.md#time-varying-collections
+[Codec]: mz_persist_types::Codec
+[Codec64]: mz_persist_types::Codec64
+
+Along with talking to the outside world, STORAGE pilots persist shards around to
+present a "STORAGE collection", which is a durable TVC that handles reshardings
+and version upgrades. A persist shard is exactly a [storage shard] and they can be used
+interchangeably.
+
+[storage shard]: https://github.com/MaterializeInc/materialize/blob/e8c27d00cda02b1f9a50fbfc852d59998b936232/doc/developer/platform/architecture-storage.md#shards
```

Comment:
I'm guessing you pointed some of these links at a commit and some at `main` intentionally? Just double checking.